main.tf

```diff
@@ -0,0 +1,114 @@
+terraform {
+
+  backend "azurerm" {
+    resource_group_name  = "bearxapp_ProjectExercise"
+    storage_account_name = "storageaccounttfstate27"
+    container_name       = "containertfstate27"
+    key                  = "terraform.tfstate"
+  }
+
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = ">= 3.8"
+    }
+  }
+}
+
+provider "azurerm" {
+  features {}
+  
+  subscription_id = var.azure_subscription_id
+  client_id       = var.azure_client_id
+  client_secret   = var.azure_client_secret
+  tenant_id       = var.azure_tenant_id
+}
+
+data "azurerm_resource_group" "main" {
+    name = "bearxapp_ProjectExercise"
+
+}
+
+resource "random_string" "resource_code" {
+  length  = 5
+  special = false
+  upper   = false
+}
+
+resource "azurerm_service_plan" "main" {
+    name = "bears-app-service-plan-tr" 
+    location = data.azurerm_resource_group.main.location 
+    resource_group_name = data.azurerm_resource_group.main.name 
+    os_type = "Linux"
+    sku_name = "B1"
+}
+
+resource "azurerm_linux_web_app" "main" {
+    name = "bearxapp-tr" 
+    location = data.azurerm_resource_group.main.location 
+    resource_group_name = data.azurerm_resource_group.main.name 
+    service_plan_id = azurerm_service_plan.main.id 
+
+ site_config { 
+    application_stack { 
+        docker_image = "slipsreamamx/todo-app" 
+        docker_image_tag = "prod-latest" 
+    } 
+ } 
+
+ app_settings = { 
```

Comment:
The indenting here is slightly confusing - running `terraform fmt` from the same folder should fix that up for you